.github/workflows/upstream-compat.yaml

```diff
@@ -0,0 +1,39 @@
+on:
+  push:
+    branches-ignore:
+    - main
+    - for-upstream/*
+
+jobs:
+  make-upstream-branch:
+    name: Make branch rebased onto upstream
+    runs-on: ubuntu-latest
+    container:
+      image: bitnami/git:2.34.1
```

Comment:
I don't have anything against using a container here. 
However `git` is one of the tools that is guaranteed to be present on the runner. 
see [here](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md)

Reply:
this is me just learning github actions then - you "always" (well, there is shell runners in theory) need containers in gitlab ci and that's what I'm used to x)
Thanks for pointing this out :)